crates/compiler/src/llvm/data_layout.rs

```diff
@@ -0,0 +1,1415 @@
+            specs.push(IntegerLayout {
+                size:                8,
+                abi_alignment:       8,
+                preferred_alignment: 8,
+            });
+        }
+
+        // Add the default for i16
+        if !specs.iter().any(|s| s.size == 16) {
+            specs.push(IntegerLayout {
+                size:                16,
+                abi_alignment:       16,
+                preferred_alignment: 16,
+            });
+        }
+
+        // Add the default for i32
+        if !specs.iter().any(|s| s.size == 32) {
+            specs.push(IntegerLayout {
+                size:                32,
+                abi_alignment:       32,
+                preferred_alignment: 32,
+            });
+        }
+
+        // Add the default for i64
+        if !specs.iter().any(|s| s.size == 64) {
+            specs.push(IntegerLayout {
+                size:                64,
+                abi_alignment:       32,
+                preferred_alignment: 64,
+            });
+        }
+
+        specs.sort();
+        specs
+    }
+
+    /// Augments the parsed pointer specifications with any missing information
+    /// based on the defaults for LLVM's data layout.
+    fn pointer_specs_or_defaults(mut specs: Vec<PointerLayout>) -> Vec<PointerLayout> {
+        if !specs.iter().any(|l| l.address_space == 0) {
+            specs.push(PointerLayout {
+                address_space:       0,
+                size:                64,
+                abi_alignment:       64,
+                preferred_alignment: 64,
+                index_size:          64,
+            });
+        }
+
+        specs.sort();
+        specs
+    }
+}
+
+impl Default for DataLayout {
+    fn default() -> Self {
+        Self::new("").expect("The empty string was not a valid data layout specification")
+    }
+}
+
+impl TryFrom<&str> for DataLayout {
+    type Error = Error;
+
+    fn try_from(value: &str) -> std::result::Result<Self, Self::Error> {
+        Self::new(value)
+    }
+}
+
+impl TryFrom<String> for DataLayout {
+    type Error = Error;
+
+    fn try_from(value: String) -> std::result::Result<Self, Self::Error> {
+        Self::new(&value)
+    }
+}
+
+/// A description of the endianness used when laying out data.
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub enum Endianness {
+    /// Little-endian (least-significant byte first).
+    Little,
+
+    /// Big-endian (most-significant byte first).
+    Big,
+}
+
+impl Endianness {
+    /// Parses the endianness specification part of the data-layout.
+    fn parser() -> impl parsing::DLParser<Endianness> {
+        choice((
+            just("e").to(Endianness::Little),
+            just("E").to(Endianness::Big),
+        ))
+    }
+}
+
+/// A description of the mangling scheme used by this module.
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub enum Mangling {
+    /// The Windows COFF mangling scheme.
```

Comment:
*pushes up non-existent glasses for a wild ultranitpick you should totally ignore*

 ~~what you're referring to as linux is, in fact GN--~~, er, I mean

the COFF mangling scheme strictly belongs to Unix COFF file format that came from SysV; its mangling scheme just happens to still be used by Windows since Portable Executable is extended from COFF